### ISSUE TITLE
Decimal value 0 is removed when flattenAsMap

### ISSUE THREAD

**Issue by SonyGeorge** [wnameless/json-flattener#84]
String jsona = "{ \"a\" : { \"b\" : 1, \"c\": null, \"d\": [false, true] }, \"e\": \"f\", \"g\":2.3 }";
        Map<String, Object> flattenJsona = JsonFlattener.flattenAsMap(jsona);
        System.out.println(flattenJsona);
output : {"a.b":1,"a.c":null,"a.d[0]":false,"a.d[1]":true,"e":"f","g":2.3}

        String jsonb = "{ \"a\" : { \"b\" : 1, \"c\": null, \"d\": [false, true] }, \"e\": \"f\", \"g\":2.35 }";
        Map<String, Object> flattenJsonb = JsonFlattener.flattenAsMap(jsonb);
        System.out.println(flattenJsonb);
output : {"a.b":1,"a.c":null,"a.d[0]":false,"a.d[1]":true,"e":"f","g":2.35}

        String jsonc = "{ \"a\" : { \"b\" : 1, \"c\": null, \"d\": [false, true] }, \"e\": \"f\", \"g\":2.30 }";
        Map<String, Object> flattenJsonc = JsonFlattener.flattenAsMap(jsonc);
        System.out.println(flattenJsonc);
output : {"a.b":1,"a.c":null,"a.d[0]":false,"a.d[1]":true,"e":"f","g":2.3}  // 0 is lost

        String jsond = "{ \"a\" : { \"b\" : 1, \"c\": null, \"d\": [false, true] }, \"e\": \"f\", \"g\":2.309 }";
        Map<String, Object> flattenJsond = JsonFlattener.flattenAsMap(jsond);
        System.out.println(flattenJsond);
output : {"a.b":1,"a.c":null,"a.d[0]":false,"a.d[1]":true,"e":"f","g":2.309}

**Comment by SonyGeorge** [comment]
https://github.com/wnameless/json-flattener/blob/master/src/main/java/com/github/wnameless/json/flattener/JsonFlattener.java#L430

https://github.com/wnameless/json-base/blob/master/src/main/java/com/github/wnameless/json/base/JsonValueBase.java#L132

**Comment by wnameless** [comment]
This issue has been solved in version: 0.16.6.
However, the default JacksonJsonCore used in JsonFlattener intends to emit trailing zeros of float numbers.
To preserve scales of float numbers, you have to use GsonJsonCore or do some configuration with ObjectMapper:
```java
 String json = "...";

 JsonNodeFactory f = new JsonNodeFactory(true);
 ObjectMapper mapper = new ObjectMapper();
 mapper.enable(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS);
 mapper.setNodeFactory(f);
 JsonFlattener jsonFlattener = new JsonFlattener(new JacksonJsonCore(mapper), json);
```
